### PR TITLE
added prompt to install language-server extension

### PR DESCRIPTION
**Pull request by daimor**
@gjsjohnmurray I need a better message for the prompt

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -342,7 +342,7 @@ async function serverManager(): Promise<any> {
     }
     await vscode.window
       .showInformationMessage(
-        "The [InterSystems® Server Manager extension](https://marketplace.visualstudio.com/items?itemName=intersystems-community.servermanager) is recommended to help you [define connections and store passwords securely](https://intersystems-community.github.io/vscode-objectscript/configuration/#configuring-a-server) in your keychain.",
+        `The [InterSystems® Server Manager extension](https://marketplace.visualstudio.com/items?itemName=${extId}) is recommended to help you [define connections and store passwords securely](https://intersystems-community.github.io/vscode-objectscript/configuration/#configuring-a-server) in your keychain.`,
```
Comment (gjsjohnmurray) [inline src/extension.ts:345]:
```suggestion
        `The [InterSystems Server Manager extension](https://marketplace.visualstudio.com/items?itemName=${extId}) is recommended to help you [define connections and store passwords securely](https://intersystems-community.github.io/vscode-objectscript/configuration/#configuring-a-server) in your keychain.`,
```

Remove the &reg; mark to align with recent changes in that extension by @isc-rsingh

File: src/extension.ts [hunk 2]
```diff
@@ -370,9 +370,35 @@ async function serverManager(): Promise<any> {
...
+  if (!extension) {
+    try {
+      await vscode.commands.executeCommand("extension.open", extId);
+    } catch (ex) {
+      // Such command do not exists, suppose we are under Theia, it's not possible to install this extension this way
+      return;
+    }
+    await vscode.window
+      .showInformationMessage(
+        `The [InterSystems® Language-Server extension](https://marketplace.visualstudio.com/items?itemName=${extId}) is recommended.`,
```
Comment (gjsjohnmurray) [inline src/extension.ts:386]:
```suggestion
        `Install the [InterSystems Language-Server extension](https://marketplace.visualstudio.com/items?itemName=${extId}) for best handling of ObjectScript code.`,
```

File: src/extension.ts [hunk 3]
```diff
@@ -534,7 +560,7 @@ export async function activate(context: vscode.ExtensionContext): Promise<any> {
     outputChannel.show(true);
   }
 
-  const languageServerExt = languageServer();
+  const languageServerExt = await languageServer();
   const noLSsubscriptions: { dispose(): any }[] = [];
   if (!languageServerExt) {
     outputChannel.appendLine(`The language-server extension was not found.\n`);
```
Comment (gjsjohnmurray) [inline src/extension.ts:566]:
```suggestion
    outputChannel.appendLine(`The intersystems.language-server extension is not installed or has been disabled.\n`);
```

**Review by gjsjohnmurray — CHANGES_REQUESTED** [review]

**Review by daimor — COMMENTED** [review]
.

**Review by gjsjohnmurray — APPROVED** [review]

**Comment by gjsjohnmurray** [comment]
@daimor is CI going to fail until language-server gets published? Or is something else broken?

**Comment by daimor** [comment]
> @daimor is CI going to fail until language-server gets published? Or is something else broken?

It fails due to waiting too much for the answer in the prompt. ignoring now in Tests

File: src/extension.ts [hunk 4]
```diff
@@ -370,9 +370,35 @@ async function serverManager(): Promise<any> {
...
+  if (!extension) {
+    try {
+      await vscode.commands.executeCommand("extension.open", extId);
+    } catch (ex) {
+      // Such command do not exists, suppose we are under Theia, it's not possible to install this extension this way
+      return;
+    }
+    await vscode.window
+      .showInformationMessage(
+        `Install the [InterSystems Language-Server extension](https://marketplace.visualstudio.com/items?itemName=${extId}) for best handling of ObjectScript code.`,
```
Comment (gjsjohnmurray) [inline src/extension.ts:386]: Remove the dash:
```suggestion
        `Install the [InterSystems Language Server extension](https://marketplace.visualstudio.com/items?itemName=${extId}) for best handling of ObjectScript code.`,
```

**Review by gjsjohnmurray — CHANGES_REQUESTED** [review]

**Review by gjsjohnmurray — APPROVED** [review]

**Comment by gjsjohnmurray** [comment]
I tested the dev vsix and it worked as expected. @daimor I will leave it to you to merge when you are ready.

**Pull request merged**